scripts/update_gpu_ids.py

```
@@ -0,0 +1,340 @@
+    r"\bEthernet\b",
+    r"\bNetwork\b",
+]
+_NON_GPU_RE = re.compile("|".join(NON_GPU_PATTERNS), re.IGNORECASE)
+
+
+def fetch_pci_ids(source: str = "github") -> str:
+    """Download the pci.ids database."""
+    url = SOURCES.get(source, source)
+    print(f"Fetching PCI IDs from {url}...")
+
+    req = urllib.request.Request(url, headers={"User-Agent": "nvidia-driver-installer/update_gpu_ids"})
```

Ruff (0.15.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)